software/contrib/noddy_holder.py

```diff
@@ -0,0 +1,162 @@
+
+digital_in: gate
+analog_in: cv
+
+knob_1: Not used
+knob_2: Not used
+
+button_1: Not used
+button_2: Not used
+
+output_1: gate 
+output_2: s&h
+output_3: t&h
+output_4: inverted gate
+output_5: s&h using inverted gate
+output_6: t&h using inverted gate
+
+'''
+# Version number
+
+VERSION="1.0"
+
+# Overclock the Pico for improved performance.
+machine.freq(250_000_000)
+
+class NoddyHolder(EuroPiScript):
+    def __init__(self):
+        self.gate = False
+        # Keep track of values for display, S&H and T&H for each
+        # channel
+        self.channel_1 = [0,-1]
+        self.channel_2 = [0,-1]
+        
+        # Triggered when button 1 is released
+        # Short press: 
+        # Long press: 
+        @b1.handler_falling
+        def b1Pressed():
+            if ticks_diff(ticks_ms(), b1.last_pressed()) >  300:
+                # long press
+                pass
+            else:
+                # short press
+                pass
+
+        # Triggered when button 2 is released.
+        # Short press: 
+        # Long press: 
+        @b2.handler_falling
+        def b2Pressed():
+            
+            if ticks_diff(ticks_ms(), b2.last_pressed()) >  300:
+                # long press
+                pass
+            else:
+                # short press
+                pass
+
+        # Triggered when din goes HIGH.
+        @din.handler
+        def dinTrigger():
+            self.gate = True
+            # Sample input
+            sample = ain.read_voltage()
+            # Set mirrored gate HIGH
+            cv1.on()
+            # Set inverse gate LOW
+            cv4.off()
+
+            # Set S&H to sample
+            cv2.voltage(sample)
+            self.channel_1[0] = sample
+            # Channel 1 is now tracking
+            
+            self.channel_1[1] = -1
+            # Set T&H of inverted gate to sample
+            cv6.voltage(sample)
+            self.channel_2[1] = sample
+            self.update_screen()
+
+
+        @din.handler_falling
+        def dinTriggerEnd():
+            self.gate = False
+            # Sample input
+            sample = ain.read_voltage()
+            # Set mirrored gate LOW
+            cv1.off()
+            # Set inverse gate HIGH
+            cv4.on()
+
+            # Set T&H of gate to sample
+            cv3.voltage(sample)
+            self.channel_1[1] = sample
+            # Set S&H of inverted gate to sample
+            cv5.voltage(sample)
+            self.channel_2[0] = sample
+            # Channel 2 is now tracking
+            self.channel_2[1] = -1
+            self.update_screen()
+
+    def update(self):
+        # Sample input
+        sample = ain.read_voltage()
+        if self.gate:
+            # Pass sample to T&H
+            cv3.voltage(sample)
+        else:
+            # Pass sample to inverted T&H 
+            cv6.voltage(sample)
+
+    def main(self):
+        while True:
+            self.update()
```

Comment:
You can call `update_screen()` once before entering the main while loop, otherwise the screen will be blank until the digital input receives its first trigger/gate.

```suggestion
    def main(self):
        self.update_screen()
        while True:
            self.update()
```

Reply:
Good catch. Thanks.